src/shared/services/fields/fields.purger.ts

```diff
@@ -35,10 +36,10 @@ export class FieldsPurger {
     currentPage.wizard_page_fields.forEach(wpf => {
       let case_field = this.findCaseFieldByWizardPageFieldId(currentPage, wpf);
       if (this.hasShowConditionField(case_field, formFields)) {
-          let condition = new ShowCondition(case_field.show_condition);
-          if (this.isHidden(condition, formFields)) {
-            this.resetField(form, case_field);
-          }
+        let condition = new ShowCondition(case_field.show_condition);
+        if (this.isHidden(condition, formFields) && !(this.isReadonly(case_field))) {
```

Comment:
can you move isReadonly method to CaseField class itself please. And add a unit test for it.

Reply:
Added comment on the function signature.